src/components/TransactionBlock/PlrDaoStakingTransactionBlock.tsx

```diff
@@ -0,0 +1,632 @@
+import React, { useCallback, useEffect, useMemo, useState } from 'react';
+import styled, { useTheme } from 'styled-components';
+import { AccountTypes, ExchangeOffer, NftList } from 'etherspot';
+import { TokenListToken } from 'etherspot/dist/sdk/assets/classes/token-list-token';
+import { ethers } from 'ethers';
+import debounce from 'debounce-promise';
+
+// Types
+import { IPlrDaoStakingMembershipBlock } from '../../types/transactionBlock';
+
+import TextInput from '../TextInput';
+import SelectInput, { SelectOption } from '../SelectInput/SelectInput';
+import { useEtherspot, useTransactionBuilder } from '../../hooks';
+import {
+  formatAmountDisplay,
+  formatMaxAmount,
+} from '../../utils/common';
+import {
+  addressesEqual,
+  isValidEthereumAddress,
+  isValidAmount,
+} from '../../utils/validation';
+import AccountSwitchInput from '../AccountSwitchInput';
+import NetworkAssetSelectInput from '../NetworkAssetSelectInput';
+import {
+  Chain,
+  supportedChains,
+  plrDaoAsset,
+  plrDaoMemberNFT,
+} from '../../utils/chain';
+import { swapServiceIdToDetails } from '../../utils/swap';
+import { IAssetWithBalance } from '../../providers/EtherspotContextProvider';
+import { RoundedImage } from '../Image';
+import { Theme } from '../../utils/theme';
+import { DestinationWalletEnum } from '../../enums/wallet.enum';
+import Text from '../Text/Text';
+export interface IPlrDaoTransactionBlockValues {
+  hasEnoughPLR: boolean;
+  fromChainId?: number;
+  fromAssetAddress?: string;
+  fromAssetIconUrl?: string;
+  fromAsset?: IAssetWithBalance;
+  selectedAsset?: IAssetWithBalance | null;
+  toAsset?: TokenListToken;
+  fromAssetDecimals?: number;
+  fromAssetSymbol?: string;
+  amount?: string;
+  accountType: AccountTypes;
+  receiverAddress?: string;
+  offer?: ExchangeOffer;
+}
+
+const Title = styled.h3`
+  margin: 0 0 18px;
+  padding: 0;
+  font-size: 16px;
+  color: ${({ theme }) => theme.color.text.cardTitle};
+  font-family: 'PTRootUIWebBold', sans-serif;
+`;
+
+const WalletReceiveWrapper = styled.div`
+  display: flex;
+  flex-direction: row;
+`;
+
+const OfferDetails = styled.div`
+  display: flex;
+  flex-direction: row;
+  align-items: center;
+  font-family: 'PTRootUIWebMedium', sans-serif;
+`;
+
+const Container = styled.div`
+  align-items: center;
+  font-family: 'PTRootUIWebMedium', sans-serif;
+  background: #21002e;
+  color: #fefefe;
```

Comment:
I understand that this has been lifted from the theme but we need to try and accmmodate the custom theming props. Have a chat with Max if you're struggling.